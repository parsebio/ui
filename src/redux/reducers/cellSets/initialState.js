const initialState = {
  properties: {},
  hierarchy: [],
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
