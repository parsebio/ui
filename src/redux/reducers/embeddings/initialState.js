/**
 * The store has the following format:
 * {
 *    embeddingType: {
 *      data: {xValues:[1,2], yValues:[5,3], cellIds:[0,3]},
 *      loading: false,
 *      error: false
 *    }
 * }
 */

const initialEmbeddingState = {
  data: {},
  loading: true,
  error: false,
};

const initialState = {};

export { initialEmbeddingState };
export default initialState;
