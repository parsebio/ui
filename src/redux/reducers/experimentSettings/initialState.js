const metaInitialState = {
  loading: false,
  completingStepError: false,
  loadingSettingsError: false,
  changedQCFilters: new Set(),
};

const initialState = {
  info: {
    experimentId: null,
    experimentName: null,
    sampleIds: [],
    pipelineVersion: null,
  },
  processing: {
    meta: metaInitialState,
  },
  originalProcessing: {},
};

export { metaInitialState };
export default initialState;
