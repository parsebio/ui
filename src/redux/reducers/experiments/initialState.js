const experimentTemplate = {
  name: null,
  description: 'Analysis description',
  id: null,
  createdAt: null,
  updatedAt: null,
  notifyByEmail: true,
  sampleIds: [],
  metadataKeys: [],
  pipelineVersion: 1,
  parentExperimentId: null,
  cellLevelMetadata: null,
  isSubsetted: false,
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeExperimentId: null,
  },
  ids: [],
};

const METADATA_DEFAULT_VALUE = 'N.A.';

export default initialState;
export { experimentTemplate, METADATA_DEFAULT_VALUE };
