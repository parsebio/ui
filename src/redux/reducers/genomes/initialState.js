const initialState = {
  loading: false,
  error: false,
  public: {},
  custom: {},
};

export default initialState;
