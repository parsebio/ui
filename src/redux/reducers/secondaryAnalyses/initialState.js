const secondaryAnalysisTemplate = {
  name: null,
  id: null,
  createdAt: null,
  kit: null,
  chemistryVersion: null,
  numOfSamples: null,
  numOfSublibraries: null,
  files: { data: {} },
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeSecondaryAnalysisId: null,
  },
  ids: [],
};

export { secondaryAnalysisTemplate };
export default initialState;
