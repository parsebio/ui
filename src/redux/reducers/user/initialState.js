const initialState = {
  current: null,
  isAuthenticated: false,
};

export default initialState;
