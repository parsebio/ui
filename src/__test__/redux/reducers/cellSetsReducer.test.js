import { enableMapSet } from 'immer';
import cellSetsReducer from 'redux/reducers/cellSets';
import initialState from 'redux/reducers/cellSets/initialState';

import {
  CELL_SETS_LOADING, CELL_SETS_LOADED,
  CELL_SETS_CREATE,
  CELL_SETS_UPDATE_PROPERTY,
  CELL_SETS_DELETE,
  CELL_SETS_ERROR,
  CELL_SETS_HIDE, CELL_SETS_UNHIDE_ALL, CELL_SETS_UNHIDE, CELL_SETS_REORDER,
} from 'redux/actionTypes/cellSets';

enableMapSet();

describe('cellSetsReducer', () => {
  it('Reduces identical state on unknown action', () => expect(
    cellSetsReducer(undefined, {
      action: 'well/this/is/not/a/valid/action',
      payload: {},
    }),
  ).toEqual(initialState));

  it('Sets loading state on loading action', () => {
    const newState = cellSetsReducer({ ...initialState, loading: false }, {
      type: CELL_SETS_LOADING,
    });

    expect(newState.loading).toEqual(true);
  });

  it('Sets up store properly for loaded cell sets.', () => {
    const data = [
      {
        key: '1',
        name: 'parent 1',
        rootNode: true,
        children: [
          {
            key: '1a',
            name: 'first child',
            color: '#00FF00',
          },
        ],
      },
      {
        key: '2',
        name: 'parent 2',
        rootNode: true,
        children: [],
      },
    ];

    const newState = cellSetsReducer(initialState, {
      type: CELL_SETS_LOADED,
      payload: {
        data,
      },
    });

    expect(newState.loading).toEqual(false);
    expect(newState.hierarchy).toEqual([{ key: '1', children: [{ key: '1a' }] }, { key: '2', children: [] }]);
    expect(newState.properties).toMatchSnapshot();
  });

  it('Adds cells to scratchpad on creation', () => {
    const data = {
      key: 'key2',
      name: 'My Key',
      color: '#ff00ff',
      cellIds: [1, 2, 3],
      type: 'cellSets',
    };

    const newState = cellSetsReducer({ ...initialState, hierarchy: [{ key: 'scratchpad', children: [] }] }, {
      type: CELL_SETS_CREATE,
      payload: data,
    });

    expect(newState.hierarchy).toMatchSnapshot();
    expect(newState.properties).toMatchSnapshot();
  });

  it('Updates cell set properties', () => {
    const state = {
      ...initialState,
      properties: {
        1: {
          name: 'parent 1',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        2: {
          name: 'parent 2',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        '1a': {
          name: 'first child',
          color: '#00FF00',
          cellIds: undefined,
          rootNode: undefined,
        },
      },
      hierarchy: [{ key: '1', children: [{ key: '1a' }] }, { key: '2', children: [] }],
    };

    const newState = cellSetsReducer(state, {
      type: CELL_SETS_UPDATE_PROPERTY,
      payload: {
        cellSetKey: '1a',
        dataUpdated: { name: 'favorite child', color: '#ffffff' },
      },
    });

    expect(newState.properties['1a']).toMatchSnapshot();
  });

  it('Removes child correctly', () => {
    const state = {
      ...initialState,
      properties: {
        1: {
          name: 'parent 1',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        2: {
          name: 'parent 2',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        '1a': {
          name: 'first child',
          color: '#00FF00',
          cellIds: undefined,
          rootNode: false,
          parentNodeKey: '1',
        },
      },
      hierarchy: [{ key: '1', children: [{ key: '1a' }] }, { key: '2', children: [] }],
    };

    const newState = cellSetsReducer(state, {
      type: CELL_SETS_DELETE,
      payload: {
        key: '1a',
      },
    });

    expect(Object.keys(newState.properties).length).toEqual(2);
    expect(newState.hierarchy.length).toEqual(2);
    expect(newState.hierarchy).toMatchSnapshot();
  });

  it('Sets error conditions', () => {
    const newState = cellSetsReducer(initialState, {
      type: CELL_SETS_ERROR,
      payload: {
        error: 'asdsadsa',
      },
    });

    expect(newState.error).toEqual('asdsadsa');
  });

  it('Adds hidden cell sets to set', () => {
    const newState = cellSetsReducer(initialState, {
      type: CELL_SETS_HIDE,
      payload: {
        key: 'new',
      },
    });

    expect(newState.hidden).toMatchSnapshot();
  });

  it('Handles rehiding of previously hidden cell sets', () => {
    const previousState = {
      ...initialState,
      hidden: new Set('a', 'b', 'c'),
    };

    const newState = cellSetsReducer(previousState, {
      type: CELL_SETS_HIDE,
      payload: {
        key: 'a',
      },
    });

    expect(newState.hidden).toMatchSnapshot();
  });

  it('Handles unhiding of previously hidden cell sets', () => {
    const previousState = {
      ...initialState,
      hidden: new Set('a', 'b', 'c'),
    };

    const newState = cellSetsReducer(previousState, {
      type: CELL_SETS_UNHIDE,
      payload: {
        key: 'a',
      },
    });

    expect(newState.hidden).toMatchSnapshot();
  });

  it('Handles unhiding of a not hidden cell set', () => {
    const previousState = {
      ...initialState,
      hidden: new Set('a', 'b', 'c'),
    };

    const newState = cellSetsReducer(previousState, {
      type: CELL_SETS_UNHIDE,
      payload: {
        key: 'e',
      },
    });

    expect(newState.hidden).toMatchSnapshot();
  });

  it('Handles unhiding all', () => {
    const previousState = {
      ...initialState,
      hidden: new Set('a', 'b', 'c'),
    };

    const newState = cellSetsReducer(previousState, {
      type: CELL_SETS_UNHIDE_ALL,
      payload: {},
    });

    expect(newState.hidden).toEqual(initialState.hidden);
  });

  it('Handles unhiding all when no cell set is hidden', () => {
    const newState = cellSetsReducer(initialState, {
      type: CELL_SETS_UNHIDE_ALL,
      payload: {},
    });

    expect(newState.hidden).toEqual(initialState.hidden);
  });

  it('reorders correctly', () => {
    const state = {
      ...initialState,
      properties: {
        1: {
          name: 'parent 1',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        2: {
          name: 'parent 2',
          color: undefined,
          cellIds: undefined,
          rootNode: true,
        },
        '1a': {
          name: 'first child',
          color: '#00FF00',
          cellIds: [],
          rootNode: false,
        },
        '1b': {
          name: 'second child',
          color: '#00FFFF',
          cellIds: [],
          rootNode: false,
        },
      },
      hierarchy: [{ key: '1', children: [{ key: '1a' }, { key: '1b' }] }, { key: '2', children: [] }],
    };

    const newState = cellSetsReducer(state, {
      type: CELL_SETS_REORDER,
      payload: {
        cellSetKey: '1a',
        newPosition: '1',
        cellClassKey: '1',
      },
    });

    expect(newState.hierarchy[0].children[1]).toEqual({ key: '1a' });

    expect(newState).toMatchSnapshot();
  });
});
