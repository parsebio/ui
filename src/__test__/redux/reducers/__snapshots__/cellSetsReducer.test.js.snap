// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cellSetsReducer Adds cells to scratchpad on creation 1`] = `
[
  {
    "children": [
      {
        "key": "key2",
      },
    ],
    "key": "scratchpad",
  },
]
`;

exports[`cellSetsReducer Adds cells to scratchpad on creation 2`] = `
{
  "key2": {
    "cellIds": Set {
      1,
      2,
      3,
    },
    "color": "#ff00ff",
    "key": "key2",
    "name": "My Key",
    "parentNodeKey": "scratchpad",
    "type": "cellSets",
  },
}
`;

exports[`cellSetsReducer Adds hidden cell sets to set 1`] = `
Set {
  "new",
}
`;

exports[`cellSetsReducer Handles rehiding of previously hidden cell sets 1`] = `
Set {
  "a",
}
`;

exports[`cellSetsReducer Handles unhiding of a not hidden cell set 1`] = `
Set {
  "a",
}
`;

exports[`cellSetsReducer Handles unhiding of previously hidden cell sets 1`] = `Set {}`;

exports[`cellSetsReducer Removes child correctly 1`] = `
[
  {
    "children": [],
    "key": "1",
  },
  {
    "children": [],
    "key": "2",
  },
]
`;

exports[`cellSetsReducer Sets up store properly for loaded cell sets. 1`] = `
{
  "1": {
    "cellSetKeys": undefined,
    "color": undefined,
    "name": "parent 1",
    "rootNode": true,
    "type": undefined,
  },
  "1a": {
    "cellSetKeys": undefined,
    "color": "#00FF00",
    "name": "first child",
    "parentNodeKey": "1",
    "rootNode": undefined,
    "type": undefined,
  },
  "2": {
    "cellSetKeys": undefined,
    "color": undefined,
    "name": "parent 2",
    "rootNode": true,
    "type": undefined,
  },
}
`;

exports[`cellSetsReducer Updates cell set properties 1`] = `
{
  "cellIds": undefined,
  "color": "#ffffff",
  "name": "favorite child",
  "rootNode": undefined,
}
`;

exports[`cellSetsReducer reorders correctly 1`] = `
{
  "error": false,
  "hidden": Set {},
  "hierarchy": [
    {
      "children": [
        {
          "key": "1b",
        },
        {
          "key": "1a",
        },
      ],
      "key": "1",
    },
    {
      "children": [],
      "key": "2",
    },
  ],
  "initialLoadPending": true,
  "loading": false,
  "properties": {
    "1": {
      "cellIds": undefined,
      "color": undefined,
      "name": "parent 1",
      "rootNode": true,
    },
    "1a": {
      "cellIds": [],
      "color": "#00FF00",
      "name": "first child",
      "rootNode": false,
    },
    "1b": {
      "cellIds": [],
      "color": "#00FFFF",
      "name": "second child",
      "rootNode": false,
    },
    "2": {
      "cellIds": undefined,
      "color": undefined,
      "name": "parent 2",
      "rootNode": true,
    },
  },
  "updatingClustering": false,
}
`;
